### ISSUE TITLE
How to override `default_max_page_size` in a connection to get infinite results?

### ISSUE THREAD

**Issue by nicolasrouanne** [rmosolgo/graphql-ruby#2840]
**Describe the bug**

I set a `default_max_page_size` at the _Schema_ level. However, on some fields, I want the connection to return **all** the results for a connection. Setting a `max_page_size` on a _Connection_ type field can **change** the page size for the connection on this field, but **not remove it**.

**Versions**

`graphql`: 1.10.3
`rails` : 6.0.2.1
`graphql-batch`: 0.4.2

**GraphQL schema**

```ruby
class ApiSchema < GraphQL::Schema
  default_max_page_size 20
  query(Types::QueryType)

  use GraphQL::Execution::Interpreter
  use GraphQL::Analysis::AST

  use GraphQL::Pagination::Connections
end

module Types
  class QueryType < Types::BaseObject
    field :users, UserType.connection_type, null: false, max_page_size: -1

    def users
      User.all
    end
  end
end
```

**GraphQL query**

```graphql
query {
  users { 
    nodes {
      id
    }
 }
}
```

```json
{
  "data": {
    "users": {
      "nodes": {
        // always 20 users even if total number is greater
      }
    }
  }
}
```

**Expected behavior**

I would expect to be able to override the `default_max_page_size` at the field or connection level, that would also mean "remove it". 
Something like `0` or `nil` to convey "infinite number of items":
```ruby
max_page_size: 0
max_page_size: nil
```

**Actual behavior**

I can't remove it properly, using `nil` or `0` for instance, since the connection is checking the provided `max_page_size` against the `default_max_page_size` [w/ a `||` check](https://github.com/rmosolgo/graphql-ruby/blob/b5d1afb9ee7a2723a54a708839c0afd505cfddd3/lib/graphql/pagination/connection.rb#L68-L70), the `default_max_page_size` value always prevail:
```ruby
max_page_size: 0 # not working
max_page_size: nil # not working
```

I found a way around using what I consider a hack, providing `-1` value to `max_page_size`
```ruby
max_page_size: -1 # working
```

**Comment by rmosolgo** [comment]
This is a bug, `nil` is supposed to work, but the code you shared obviously breaks it 🤦‍♂ . 

Thanks for the detailed report, I'll try to take a look soon. Glad that you found a work-around in the meantime.

**Comment by nicolasrouanne** [comment]
I'm not sure if #2843 fixes this issue. Here is a reproduction of the bug on `1.10.7`.

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'
  gem 'graphql', '1.10.7'
end

USERS = [{ name: 'Yoda' }, { name: 'Anakin' }, { name: 'Obi Wan' }].freeze

module Types
end

class Types::BaseObject < GraphQL::Schema::Object
  connection_type_class GraphQL::Types::Relay::BaseConnection
end

class Types::UserType < Types::BaseObject
  field :name, String, null: true

  def name
    object[:name]
  end
end

class Types::QueryType < Types::BaseObject
  # overriding `max_page_size` with nil, doesn't seem to be taken into account
  field :users, Types::UserType.connection_type, null: true, max_page_size: nil

  def users
    USERS
  end
end

class Schema < GraphQL::Schema
  default_max_page_size 2
  query(Types::QueryType)

  use(GraphQL::Execution::Interpreter)
  use(GraphQL::Analysis::AST)

  use(GraphQL::Pagination::Connections)
end

pp Schema.execute(<<~GRAPHQL).to_h
  {
    users {
      nodes {
        name
      }
    }
  }
GRAPHQL
# => {"data"=>{"users"=>{"nodes"=>[{"name"=>"Yoda"}, {"name"=>"Anakin"}]}}}
# expected: {"data"=> {"users"=> {"nodes"=>[{"name"=>"Yoda"}, {"name"=>"Anakin"}, {"name"=>"Obi Wan"}]}}}
```

**Comment by nicolasrouanne** [comment]
@rmosolgo Any news about this? Or am I missing something? Thanks 🙏🏻

**Comment by nicolasrouanne** [comment]
@rmosolgo Could we please reopen this?

**Comment by rmosolgo** [comment]
Yes, sorry I missed this before!

**Comment by rmosolgo** [comment]
@nicolasrouanne , thanks for the replication script. I adapted it to a unit test and addressed the underlying issue in #3438. It should be on master soon.
